### PR TITLE
Stop using `QuickLens` in non-test code

### PR DESCRIPTION
**Pull request by t-bast**
`QuickLens` is handy when modifying deeply nested structures, but since it internally uses reflection, it may break deterministic builds. We thus remove all usage of this library in non-test code. We didn't really need it anyway, it's mostly in tests that we do deep modifications of channel internal state to be able to test edge cases.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/offer/DefaultOfferHandler.scala [hunk 1]
```diff
@@ -83,9 +82,13 @@ object DefaultOfferHandler {
         Behaviors.withMdc(Logs.mdc(category_opt = Some(LogCategory.PAYMENT), offerId_opt = Some(invoiceRequest.offer.offerId))) {
           Behaviors.receiveMessagePartial {
             case GetPaymentPaths(replyTo, blindedPathFirstNodeId) =>
-              val routeParams = nodeParams.routerConf.pathFindingExperimentConf.getRandomConf().getDefaultRouteParams
-                .modify(_.boundaries.maxRouteLength).setTo(nodeParams.offersConfig.paymentPathLength)
-                .modify(_.boundaries.maxCltv).setTo(nodeParams.offersConfig.paymentPathCltvExpiryDelta)
+              val defaultRouteParams = nodeParams.routerConf.pathFindingExperimentConf.getRandomConf().getDefaultRouteParams
+              val routeParams = defaultRouteParams.copy(boundaries = Router.SearchBoundaries(
+                maxFeeFlat = defaultRouteParams.boundaries.maxFeeFlat,
+                maxFeeProportional = defaultRouteParams.boundaries.maxFeeProportional,
```
Comment (sstone) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/offer/DefaultOfferHandler.scala:88]: These 2 could be removed ?

**Review by sstone — APPROVED** [review]
